src/lib/evil/evil_time.c

```diff
@@ -8,9 +8,11 @@
 
 #include "evil_private.h"
 
-inline struct tm *localtime_r(const time_t * time, struct tm * result)
+EVIL_API struct tm *
+evil_localtime_r(const time_t * time, struct tm * result)
 {
-    return NULL;
+   *result = *localtime(time);
+   return result;
```

Comment:
Hmmm wait wait, actually have another problem here. I think you'll have to first store it in a temporary `struct tm*`, because if `localtime` returns `NULL`, `*localtime` will try to derreference a `NULL`.

Maybe this?

```suggestion
   struct tm* _result = localtime(time);
   
   if (_result)
     {
        *result = *_result;
        return result; // or should we return `_result`? Are they the same pointer? How to deal?
     }
   return NULL;
```

Reply:
ok...so why not the original commit: 73e3740ca29a7983d7bccb5a30564dc9e4c83e60?

Reply:
....because it doesn't solve the problem.

Reply:
nice, could you explain?

Reply:
I already explained. It has the very same issue I commented, look closely to this line:

```c
*result = *localtime(time);
```

Now re-read:

> Hmmm wait wait, actually have another problem here. I think you'll have to first store it in a temporary `struct tm*`, because **if `localtime` returns `NULL`, `*localtime` will try to derreference a `NULL`.**

Reply:
That commit still have that very same line

Reply:
ok, that makes sense, thanks